### PR TITLE
ci: add dependabot config for enable github action auto update `AP-1514`

### PR DESCRIPTION
**Pull request by BryanttV**
### Description

This PR adds the `dependabot.yml` file to perform automated upgrading to GitHub Actions. The current configuration creates a Weekly PR (on Mondays) with the changes.

Also, the versions of the workflow actions are updated to use the MAJOR version (if possible).

### Jira Issue
[AP-1514](https://edunext.atlassian.net/browse/AP-1514)

[AP-1514]: https://edunext.atlassian.net/browse/AP-1514?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/python-publish.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       run: |
         python -m build --sdist --wheel --outdir dist/ .
     - name: Publish package
-      uses: pypa/gh-action-pypi-publish@v1.9.0
+      uses: pypa/gh-action-pypi-publish@release/v1
```
Comment (mariajgrimaldi) [inline .github/workflows/python-publish.yml:37]: I wonder if dependabot recognizes this kind of pattern. Is there any documentation available saying so?

Comment (BryanttV) [reply .github/workflows/python-publish.yml]: It seems that Dependabot only looks at semantic versioning. It is better to use a specific version in this case: https://github.com/eduNEXT/eox-tenant/commit/fbdfe1c1c0d0fc7c016b6435f9d8ae6ab11edacd

File: .github/workflows/python-publish.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
       run: |
         python -m build --sdist --wheel --outdir dist/ .
     - name: Publish package
-      uses: pypa/gh-action-pypi-publish@v1.9.0
+      uses: pypa/gh-action-pypi-publish@v1.10.3
```
Comment (magajh) [inline .github/workflows/python-publish.yml:37]: Should we use major here?

Comment (magajh) [reply .github/workflows/python-publish.yml:37]: Thanks for clarifying!

**Review by magajh — APPROVED** [review]

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**